### PR TITLE
Update the docs for 0.5

### PR DESCRIPTION
**Pull request by travisn**
- Remove links to yaml files until auto docs are generated correctly
- Add note about dataDirHostPath to survive reboots
- Add note about copying the secret to your namespace

### CODE REVIEW

File: Documentation/cluster-tpr.md [hunk 1]
```diff
@@ -75,4 +75,4 @@ A Placement configuration is specified (according to the kubernetes [PodSpec](ht
 
 ## Sample
 
-A sample cluster TPR can be found and used in the [rook-cluster.yaml](https://github.com/rook/rook/blob/master/demo/kubernetes/rook-cluster.yaml) file in the Kubernetes demo directory.
+A sample cluster TPR can be found and used in the `demo/kubernetes/rook-cluster.yaml` file.
```
Comment (ilovett) [inline Documentation/cluster-tpr.md]: compared to the other demo links, this one does not have a prepended `/`

**Comment by jbw976** [comment]
I'm seeing a header table on each page with columns for "title | weight | indent".  example: https://github.com/travisn/rook/blob/clarify-docs/Documentation/advanced-configuration.md

is that expected?

**Comment by travisn** [comment]
yes, the header is actually expected. It's needed for generating docs.

File: Documentation/k8s-block.md [hunk 2]
```diff
@@ -14,10 +14,33 @@ This guide assumes you have created a Rook cluster as explained in the main [Kub
 
 ## Provision Storage
 
-Before Rook can start provisioning storage, a StorageClass and its storage pool need to be created. This is needed for Kubernetes to interoperate with Rook for provisioning persistent volumes. The [rook-storageclass.yaml](https://github.com/rook/rook/blob/master/demo/kubernetes/rook-storageclass.yaml) sample will create the storage pool automatically. For more options on pools, see the documentation on [creating storage pools](pool-tpr.md).
+Before Rook can start provisioning storage, a StorageClass and its storage pool need to be created. This is needed for Kubernetes to interoperate with Rook for provisioning persistent volumes. For more options on pools, see the documentation on [creating storage pools](pool-crd.md).
+
```
Comment (jbw976) [inline Documentation/k8s-block.md]: we still have instructions below in the "Consume the storage" section to do things like `kubectl create -f mysql.yaml`.  Should we be doing that here too for the rook-storageclass.yaml file that we inlined?

File: Documentation/k8s-object.md [hunk 3]
```diff
@@ -54,9 +54,26 @@ rook-ceph-rgw   10.3.0.177   <none>        53390/TCP   2m
 ```
 
 Now create the external service:
-```bash
-cd demo/kubernetes
-kubectl create -f rgw-external.yaml
```
Comment (jbw976) [inline Documentation/k8s-object.md]: we're not being consistent about providing an instruction about how to use the inlined yaml.  in some places we say to save it as a specific file name then to `kubectl create` on that file and other places we just have the inline yaml with no specific instruction for what to do with the yaml content.

File: Documentation/kubernetes.md [hunk 4]
```diff
@@ -40,9 +40,29 @@ kubectl create -f rook-operator.yaml
...
+  name: rook
+---
+apiVersion: rook.io/v1alpha1
+kind: Cluster
+metadata:
+  name: rook
+  namespace: rook
+spec:
+  versionTag: master
+  dataDirHostPath:
```
Comment (jbw976) [inline Documentation/kubernetes.md]: in this sample, should we have `dataDirHostPath` set to a value?

File: Documentation/k8s-block.md (hunk 2 above)
Comment (travisn) [inline Documentation/k8s-block.md]: Yeah, i'll make sure there is consistency for creating the yaml. I was assuming the `create -f` was implied, but that's not a safe assumption and I see documentation in other repos including that step as well.

File: Documentation/k8s-object.md (hunk 3 above)
Comment (travisn) [inline Documentation/k8s-object.md]: yes, will make that consistent

File: Documentation/kubernetes.md (hunk 4 above)
Comment (travisn) [inline Documentation/kubernetes.md]: Had some discussion on that in slack, we aren't setting the default to keep it simpler until we implement the k8s local storage

**Pull request merged**